### PR TITLE
Add Mock Acceptance tests for kubernetes secret resource 

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Add Mock Acceptance tests for kubernetes secret resource 

Screenshot 2023-07-06 at 10 33 50 AM

### CODE REVIEW

**Comment by Shubbhang351** [comment]
Screenshot 2023-07-06 at 10 33 50 AM

File: internal/resources/kubernetessecret/resource_secret_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+	"github.com/go-openapi/strfmt"
+	"github.com/go-test/deep"
+	"github.com/jarcoal/httpmock"
+
+	secretmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+	objectmetamodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/objectmeta"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/status"
+)
+
+func getMockSpec2() secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretSpec {
```
Comment (ishangupta-ds) [inline internal/resources/kubernetessecret/resource_secret_mock_test.go:25]: Can we use some better naming?

File: internal/resources/kubernetessecret/resource_secret_mock_test.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+		} else {
+			return nil, err
+		}
+
+		return successFunc()
+	}
+}
+
+// Register a new responder when the given call is made.
+func changeStateResponder2(registerFunc func(), successResponse int, successResponseBody interface{}) httpmock.Responder {
```
Comment (ishangupta-ds) [inline internal/resources/kubernetessecret/resource_secret_mock_test.go:84]: same as above

File: internal/resources/kubernetessecret/resource_secret_mock_test.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	return func(r *http.Request) (*http.Response, error) {
+		registerFunc()
+		return httpmock.NewJsonResponse(successResponse, successResponseBody)
+	}
+}
+
+// nolint: ineffassign,staticcheck
+func (testConfig *testAcceptanceConfig) setupHTTPMocks2(t *testing.T) {
```
Comment (ishangupta-ds) [inline internal/resources/kubernetessecret/resource_secret_mock_test.go:92]: please use a better name

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: internal/resources/kubernetessecret/resource_secret_mock_test.go (hunk 1 above)
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_mock_test.go]: updated

File: internal/resources/kubernetessecret/resource_secret_mock_test.go (hunk 2 above)
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_mock_test.go]: updated

File: internal/resources/kubernetessecret/resource_secret_mock_test.go (hunk 3 above)
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_mock_test.go]: updated

**Review by ishangupta-ds — APPROVED** [review]

File: internal/resources/kubernetessecret/resource_secret_mock_test.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+				},
+			},
+		},
+	}
+
+	getResponse := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceGetSecretResponse{
+		Secret: getModel,
+	}
+
+	postEndpoint := fmt.Sprintf("https://%s/v1alpha1/clusters/%s/namespaces/%s/secrets", endpoint, testConfig.ClusterName, testConfig.NamespaceName)
```
Comment (ramya-bangera) [inline internal/resources/kubernetessecret/resource_secret_mock_test.go:190]: Can you use the existing helper method to construct the URL as we do it in the client files?

File: internal/resources/kubernetessecret/resource_secret_test.go [hunk 5]
```diff
@@ -45,23 +50,66 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
...
+		return config.SetupWithDefaultTransportForTesting()
+	}
+
+	return config.Setup()
+}
 
+func TestAcceptanceForSecretResource(t *testing.T) {
 	testConfig := testGetDefaultAcceptanceConfig(t)
 
+	// If the flag to execute kubernetes secret tests is not found, run this as a unit test by setting up an http intercept for each endpoint
```
Comment (ramya-bangera) [inline internal/resources/kubernetessecret/resource_secret_test.go:77]: Do you mean " run this as a **mock** test by setting up an http intercept for each endpoint" ?

File: internal/resources/kubernetessecret/resource_secret_test.go [hunk 6]
```diff
@@ -45,23 +50,66 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
...
+	if _, found := os.LookupEnv("ENABLE_SECRET_ENV_TEST"); !found {
+		os.Setenv("TF_ACC", "true")
+		os.Setenv("TMC_ENDPOINT", "playground.tmc.cloud.vmware.com")
```
Comment (ramya-bangera) [inline internal/resources/kubernetessecret/resource_secret_test.go:80]: You can provide dummy TMC URL to better indicate it is mock tests

**Comment by ramya-bangera** [comment]
Can you also attach the screenshot the actual acceptance tests with this change?

File: internal/resources/kubernetessecret/resource_secret_test.go (hunk 5 above)
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_test.go]: updated PTAl

File: internal/resources/kubernetessecret/resource_secret_mock_test.go (hunk 4 above)
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_mock_test.go]: updated PTAL

File: go.mod [hunk 7]
```diff
@@ -3,14 +3,12 @@ module github.com/vmware/terraform-provider-tanzu-mission-control
...
 require (
-	github.com/go-openapi/runtime v0.19.4
 	github.com/go-openapi/strfmt v0.19.3
 	github.com/go-openapi/swag v0.19.5
-	github.com/go-openapi/validate v0.19.5
 	github.com/go-test/deep v1.0.3
 	github.com/google/go-cmp v0.5.8 // indirect
 	github.com/hashicorp/go-cty v1.4.1-0.20200414143053-d3edf31b6320
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.12.0
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.16.0
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply go.mod]: removed PTAL

File: internal/resources/kubernetessecret/resource_secret_test.go (hunk 6 above)
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_test.go]: updated PTAL

File: internal/resources/kubernetessecret/resource_secret_test.go (hunk 5 above)
Comment (Shubbhang351) [reply internal/resources/kubernetessecret/resource_secret_test.go]: updated the comment

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**